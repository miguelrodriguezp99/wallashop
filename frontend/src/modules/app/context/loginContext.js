import { createContext, useState } from "react";

// Este es el que tenemos que consumir
export const LogginContext = createContext();

// Este es el que nos provee de acceso al contexto
export function LogginProvider({ children }) {
	const [token, setToken] = useState(null);
	const [image, setImage] = useState(null);
	const [user, setUser] = useState({});

	return (
		<LogginContext.Provider
			value={{
				token,
				setToken,
				image,
				setImage,
				user,
				setUser,
			}}
		>
			{children}
		</LogginContext.Provider>
	);
}
