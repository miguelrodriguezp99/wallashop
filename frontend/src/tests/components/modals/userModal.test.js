import React from 'react';
import UserModal from '../../../modules/app/components/modals/userModal';
import { LogginContext } from '../../../modules/app/context/loginContext';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom'; 
import {render, fireEvent, waitFor, getAllByText} from '@testing-library/react';

jest.mock('../../../modules/app/firebase/config.js', () => ({
}));

const fakeLogginContext = {
    user: { id: 1 }, 
    token: "testToken",
};
  
describe("UserModal", () => {
  it("renders correctly", () => {
    const tree = renderer.create(
      <MemoryRouter>
        <LogginContext.Provider value={fakeLogginContext}>
          <UserModal />
        </LogginContext.Provider>
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
 
});