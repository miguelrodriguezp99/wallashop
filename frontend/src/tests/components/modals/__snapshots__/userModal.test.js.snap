// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserModal renders correctly 1`] = `
Array [
  <button
    onClick={[Function]}
  >
    <img
      alt="Rounded avatar"
      className="w-12 h-12 rounded-full"
    />
  </button>,
  <div
    className="mt-20"
  />,
]
`;
